<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect><x>0</x><y>0</y><width>592</width><height>380</height></rect>
  </property>
  <property name="windowTitle">
   <string>Youtube Downloader</string>
  </property>

  <widget class="QWidget" name="centralwidget">
   <!-- Complete, valid stylesheet -->
   <property name="styleSheet">
    <string notr="true">
/* Main window chrome */
QMainWindow[theme="light"] { background:#f6f6f6; color:#1a1a1a; }
QMainWindow[theme="dark"]  { background:#2b2b2b; color:#f0f0f0; }

/* Menu and status bars */
QMenuBar[theme="light"], QStatusBar[theme="light"] { background:#f6f6f6; color:#1a1a1a; }
QMenuBar[theme="dark"],  QStatusBar[theme="dark"]  { background:#2b2b2b; color:#f0f0f0; }

/* ---------- LIGHT ---------- */
#centralwidget[theme="light"] { background:#f6f6f6; color:#1a1a1a; }
#centralwidget[theme="light"] QLineEdit,
#centralwidget[theme="light"] QComboBox { background:#ffffff; border:1px solid #cfcfcf; border-radius:6px; padding:6px; }
#centralwidget[theme="light"] QPushButton { background:#ffffff; border:1px solid #cfcfcf; border-radius:8px; padding:8px 12px; font-weight:600; }
#centralwidget[theme="light"] QPushButton:hover { background:#f0f0f0; }
#centralwidget[theme="light"] QPushButton:pressed { background:#e7e7e7; }
#centralwidget[theme="light"] QPushButton:disabled { background:#f2f2f2; color:#9a9a9a; border-color:#d8d8d8; }
#centralwidget[theme="light"] QRadioButton, #centralwidget[theme="light"] QCheckBox { spacing:6px; }
/* Combo popup (light) */
#centralwidget[theme="light"] QComboBox QAbstractItemView { background:#ffffff; color:#1a1a1a; border:1px solid #cfcfcf; }

/* ProgressBar (light) */
#centralwidget[theme="light"] QProgressBar {
  border:1px solid #cfcfcf; border-radius:7px; background:#ffffff; color:#1a1a1a; text-align:center;
}
#centralwidget[theme="light"] QProgressBar::chunk { border-radius:7px; background:#3d7dff; margin:0; }

/* ---------- DARK ---------- */
#centralwidget[theme="dark"]  { background:#2b2b2b; color:#f0f0f0; }
#centralwidget[theme="dark"]  QLineEdit,
#centralwidget[theme="dark"]  QComboBox { background:#3a3a3a; border:1px solid #444; border-radius:6px; padding:6px; color:#f0f0f0; }
#centralwidget[theme="dark"]  QPushButton { background:#444; border:1px solid #555; border-radius:8px; padding:8px 12px; font-weight:600; color:#f0f0f0; }
#centralwidget[theme="dark"]  QPushButton:hover { background:#505050; }
#centralwidget[theme="dark"]  QPushButton:pressed { background:#585858; }
#centralwidget[theme="dark"]  QPushButton:disabled { background:#3a3a3a; color:#888; border-color:#444; }
#centralwidget[theme="dark"]  QRadioButton, #centralwidget[theme="dark"]  QCheckBox { spacing:6px; }
/* Combo popup (dark) */
#centralwidget[theme="dark"]  QComboBox QAbstractItemView { background:#3a3a3a; color:#f0f0f0; border:1px solid #444; }

/* ProgressBar (dark) */
#centralwidget[theme="dark"] QProgressBar {
  border:1px solid #444; border-radius:7px; background:#3a3a3a; color:#f0f0f0; text-align:center;
}
#centralwidget[theme="dark"] QProgressBar::chunk { border-radius:7px; background:#7aa2ff; margin:0; }
    </string>
   </property>

   <!-- Title -->
   <widget class="QLabel" name="label">
    <property name="geometry"><rect><x>130</x><y>10</y><width>331</width><height>41</height></rect></property>
    <property name="font"><font><family>Leelawadee UI</family><pointsize>22</pointsize><weight>75</weight><bold>true</bold></font></property>
    <property name="text"><string>Youtube Downloader</string></property>
    <property name="alignment"><set>Qt::AlignCenter</set></property>
   </widget>

   <!-- URL -->
   <widget class="QLineEdit" name="lineEdit_3">
    <property name="geometry"><rect><x>20</x><y>60</y><width>551</width><height>41</height></rect></property>
    <property name="font"><font><family>Leelawadee UI</family><pointsize>10</pointsize></font></property>
    <property name="text"><string/></property>
    <property name="placeholderText"><string>Enter URL Here...</string></property>
    <property name="alignment"><set>Qt::AlignCenter</set></property>
   </widget>

   <!-- Path -->
   <widget class="QLineEdit" name="lineEdit_4">
    <property name="geometry"><rect><x>20</x><y>110</y><width>471</width><height>41</height></rect></property>
    <property name="font"><font><family>Leelawadee UI</family><pointsize>10</pointsize></font></property>
    <property name="text"><string/></property>
    <property name="alignment"><set>Qt::AlignCenter</set></property>
   </widget>

   <!-- Set button -->
   <widget class="QPushButton" name="pushButton_4">
    <property name="geometry"><rect><x>500</x><y>110</y><width>71</width><height>41</height></rect></property>
    <property name="font"><font><family>Leelawadee UI</family><pointsize>14</pointsize><weight>75</weight><bold>true</bold></font></property>
    <property name="text"><string>Set</string></property>
   </widget>

   <!-- Download quality label -->
   <widget class="QLabel" name="label_3">
    <property name="geometry"><rect><x>50</x><y>160</y><width>151</width><height>21</height></rect></property>
    <property name="font"><font><family>Leelawadee UI</family><pointsize>12</pointsize><weight>75</weight><bold>true</bold></font></property>
    <property name="text"><string>Download Quality:</string></property>
    <property name="alignment"><set>Qt::AlignCenter</set></property>
   </widget>

   <!-- Quality combo -->
   <widget class="QComboBox" name="comboBox">
    <property name="geometry"><rect><x>90</x><y>190</y><width>69</width><height>22</height></rect></property>
    <property name="font"><font><family>Leelawadee UI</family><pointsize>10</pointsize></font></property>
    <item><property name="text"><string>Best</string></property></item>
    <item><property name="text"><string>Semi</string></property></item>
    <item><property name="text"><string>Worst</string></property></item>
   </widget>

   <!-- Download button -->
   <widget class="QPushButton" name="pushButton_3">
    <property name="geometry"><rect><x>240</x><y>170</y><width>111</width><height>51</height></rect></property>
    <property name="font"><font><family>Leelawadee UI</family><pointsize>14</pointsize><weight>75</weight><bold>true</bold></font></property>
    <property name="text"><string>Download</string></property>
   </widget>

   <!-- Single/Playlist -->
   <widget class="QRadioButton" name="radioButton">
    <property name="geometry"><rect><x>390</x><y>160</y><width>81</width><height>21</height></rect></property>
    <property name="font"><font><family>Leelawadee UI</family><pointsize>12</pointsize><weight>75</weight><bold>true</bold></font></property>
    <property name="text"><string>Single</string></property>
   </widget>
   <widget class="QRadioButton" name="radioButton_2">
    <property name="geometry"><rect><x>470</x><y>160</y><width>81</width><height>21</height></rect></property>
    <property name="font"><font><family>Leelawadee UI</family><pointsize>12</pointsize><weight>75</weight><bold>true</bold></font></property>
    <property name="text"><string>Playlist</string></property>
   </widget>

   <!-- Download Video checkbox -->
   <widget class="QCheckBox" name="checkBox">
    <property name="geometry"><rect><x>390</x><y>190</y><width>151</width><height>21</height></rect></property>
    <property name="font"><font><family>Leelawadee UI</family><pointsize>12</pointsize><weight>75</weight><bold>true</bold></font></property>
    <property name="text"><string>Download Video</string></property>
   </widget>

	<!-- Status/done label (wide, 2 lines) -->
	<widget class="QLabel" name="label_2">
	  <property name="geometry"><rect><x>16</x><y>270</y><width>560</width><height>38</height></rect></property>
	  <property name="font"><font><family>Leelawadee UI</family><pointsize>14</pointsize><weight>50</weight><bold>false</bold></font></property>
	  <property name="alignment"><set>Qt::AlignCenter</set></property>
	  <property name="wordWrap"><bool>true</bool></property>
	</widget>

	<!-- Progress label (same slot as label_2; appears during download) -->
	<widget class="QLabel" name="label_progress">
	  <property name="geometry"><rect><x>16</x><y>270</y><width>560</width><height>38</height></rect></property>
	  <property name="alignment"><set>Qt::AlignCenter</set></property>
	  <property name="wordWrap"><bool>true</bool></property>
	  <property name="visible"><bool>false</bool></property>
	</widget>

	<!-- Progress bar (centered, below the labels) -->
	<widget class="QProgressBar" name="progressBar">
	  <!-- width=300, centered => x=(592-300)/2=146 -->
	  <property name="geometry"><rect><x>146</x><y>312</y><width>300</width><height>14</height></rect></property>
	  <property name="minimum"><number>0</number></property>
	  <property name="maximum"><number>100</number></property>
	  <property name="value"><number>0</number></property>
	  <property name="textVisible"><bool>true</bool></property>
	  <property name="format"><string>%p%</string></property>
	  <property name="visible"><bool>false</bool></property>
	</widget>

	<!-- Toggle Theme (a bit lower; right-aligned) -->
	<widget class="QPushButton" name="themeButton">
	  <!-- right margin ~16 => x=592-160-16=416 -->
	  <property name="geometry"><rect><x>416</x><y>332</y><width>160</width><height>38</height></rect></property>
	  <property name="font"><font><family>Leelawadee UI</family><pointsize>12</pointsize><weight>75</weight><bold>true</bold></font></property>
	  <property name="text"><string>Toggle Theme</string></property>
	</widget>


  </widget>

  <widget class="QMenuBar" name="menubar">
   <property name="geometry"><rect><x>0</x><y>0</y><width>592</width><height>21</height></rect></property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
